### ISSUE TITLE
Allow TypeScript to infer all available extensions

### ISSUE THREAD

**Issue by ifiokjr** [remirror/remirror#680]
### Description

Currently, when using `useRemirror` it's up to the developer to add all the extension that they want to use. 

```ts
const { commands } = useRemirror(); // `commands` only has builtin auto suggestions.
```

To add automatic suggestions you need to create the type union of the currently used extensions and presets.

```ts
import { BoldExtension } from 'remirror/extension/bold';
import { CorePreset } from 'remirror/preset/core';

const { commands } = useRemirror<CorePreset | BoldExtension>();

commands.toggleBold(); // Now provides inline intellisense
commands.undo(); // No provides inline intellisense
```

The proposal which will be an experimental feature and rely on user feedback will be to automatically apply all loaded extensions to the suggested intellisense when calling `useRemirror`.

This can be done by creating a global interface `Remirror.AllExtensions`.

```ts
declare global {
  namespace Remirror {
    interface AllExtensions {
      /** Add your extension here to show up as an automatic suggestion when installed */
      custom: CustomExtension;
    }
  }
}
```

The `useRemirror` function and similar methods will uses the union of the values on this interface as the default extensions. For users wanting more control the override will still work for limiting the suggestions available.

Please note this is an **experimental feature**.

Possible issues.

- Seeing all the methods available with auto-suggest may lead people to believe that all these methods are available even if the extension isn't actually part of the current editor.
- As the number of commands grows, there may be too suggested commands and it can be harder to find what you're looking for. 

In short this might make it easier to get started, but harder to understand what's going on when things don't work.
